Makefile

```
@@ -123,12 +123,13 @@ ec/install:
 	$(@D) --version
 	@ echo "[$@]: Completed successfully!"
 
+install/pip: CUSTOM_VERSION_CMD ?= ''
```

I don't think this var is actually being used, since the target is `install/pip/%` instead of `install/pip`. But that makes me think we can simplify this to eliminate the `if` logic by setting a real default. e.g.

```suggestion
install/pip/%: PIP_PKG_VERSION_CMD ?= % --version
```

---

See line 154; CUSTOM_VERSION_CMD is used.  bumpversion would fail if you use the `--version` argument, but the version can be obtained through a different way, hence the need for an alternate, or custom, command.

---

I understand that, but that is a different target. `install/pip` is not `install/pip/%`. This particular line is not being used, as is. Either way, the suggestion is not related to that comment, particularly. We can simplify the implementation by making the var default value be `% --version`. Then bumpversion can still override it.

---

Loren, this fails.  For example:

```
[2]$ make pydocstyle/install
[install/pip/pydocstyle]: Installing pydocstyle...
python3 -m pip install --user pydocstyle
Requirement already satisfied: pydocstyle in /home/kbalk/.local/lib/python3.8/site-packages (5.1.1)
Requirement already satisfied: snowballstemmer in /home/kbalk/.local/lib/python3.8/site-packages (from pydocstyle) (2.0.0)
WARNING: You are using pip version 20.3.1; however, version 20.3.3 is available.
You should consider upgrading via the '/home/kbalk/.pyenv/versions/3.8.6/bin/python3 -m pip install --upgrade pip' command.
ln -sf ~/.local/bin/pydocstyle /home/kbalk/bin/pydocstyle
% --version
bash: line 0: fg: no job control
make[1]: *** [install/pip/pydocstyle] Error 1
make: *** [pydocstyle/install] Error 2
```

I tried the route of defining a variable and specifying a special value for bumpversion, but ran into problems whereby 'make' would override the default value with the value defined for bumpversion as it processed the contents of the makefile.

---

My bad, I misremembered how to reference the variable part of the target. I thought in the assignment like that, you just used `%`. But looks like it is still `$*`...

```suggestion
install/pip/%: PIP_PKG_VERSION_CMD ?= $* --version
```

---

Tested using:

```
$ cat Makefile
export SHELL := /bin/bash

foo/%: WUT = $*
foo/%:
        echo "test = $@"
        echo "WUT = $(WUT)"
```

```
$ make foo/bar
echo "test = foo/bar"
test = foo/bar
echo "WUT = bar"
WUT = bar
```

---

That's fine for the other installs, but not bumpversion.  The idea was to allow a custom command to produce a version number for those commands that don't support the '--version' option.  Bumpversion has an equivalent.  

I'm fine with you taking over from here and make whatever changes you wish.

---

The assignment is just a default value. The bumpversion install would still override it. But ok, I'll get to it in a few minutes.

---

ok, it's working